src/main/scala/wdl4s/Task.scala

```diff
@@ -4,26 +4,37 @@ import java.util.regex.Pattern
 
```

Comment:
Sorry for the bad diff on this file...

* Before there were two fold functions (one for checking the input `Declaration`s and one for checking output `TaskOutput`s)
* I made `TaskOutput` into a `Declaration` so now there needs to be only one fold function.
* I added `typeCheckExpression()` for checking type compatibility for all `Declaration`s
* Removed `inputs` field on `Task`.  Using `declarations` field instead.